src/components/PrimaryNav/PrimaryNav.tsx

```diff
@@ -16,23 +16,34 @@ import type { Controllers } from "store/juju/types";
 import urls, { externalURLs } from "urls";
 import "./_primary-nav.scss";
 
+type PrimaryNavLinkProps = {
+  children: React.ReactNode;
+  to: string;
+};
+
+const PrimaryNavLink = ({ children, to }: PrimaryNavLinkProps) => (
```

Comment:
I think this could also contain the icon and title as that is repeated for each item. Something like:

```jsx
const PrimaryNavLink = ({ children, to, title, icon }: PrimaryNavLinkProps) => (
  <NavLink
    className={({ isActive }) =>
      classNames("p-list__link", {
        "is-selected": isActive,
      })
    }
    to={to}
  >
    <Icon name={icon} light />
    <span className="hide-collapsed">{title}</span>
    {children}
  </NavLink>
);
```

Also, I know this file already has a few components in it, but we should try to have components in their own file so this could get moved to `src/components/PrimaryNav/PrimaryNavLink/PrimaryNavLink.tsx` and have its own tests.